### PR TITLE
Fix lint errors

### PR DESCRIPTION
**Pull request by tomc974**
This PR is just fixing a bunch of lint errors. Each error category has it's own commit so if any one of them doesn't suit your style, it will be easy to revert it.

### CODE REVIEW

File: types/priv_validator/sign_info.go [hunk 1]
```diff
@@ -149,11 +149,11 @@ func (lsi *LastSignedInfo) SignVote(signer types.Signer, chainID string, vote *t
 // If the HRS are equal and the only thing changed is the timestamp, it sets the timestamp to the previous
 // value and the Signature to the LastSignedInfo.Signature.
 // Else it returns an error.
-func (lsi *LastSignedInfo) SignProposal(signer types.Signer, chainID string, proposal *types.Proposal) error {
+func (info *LastSignedInfo) SignProposal(signer types.Signer, chainID string, proposal *types.Proposal) error {
```
Comment (melekes) [inline types/priv_validator/sign_info.go]: what's wrong with `lsi`?

File: config/config.go [hunk 2]
```diff
@@ -170,23 +170,23 @@ func (b BaseConfig) ChainID() string {
...
-func (b BaseConfig) NodeKeyFile() string {
-	return rootify(b.NodeKey, b.RootDir)
+func (c BaseConfig) NodeKeyFile() string {
+	return rootify(c.NodeKey, c.RootDir)
 }
 
 // DBDir returns the full path to the database directory
-func (b BaseConfig) DBDir() string {
-	return rootify(b.DBPath, b.RootDir)
+func (c BaseConfig) DBDir() string {
```
Comment (melekes) [inline config/config.go]: why `cfg` below, but here you are using `c`? should not we be consistent? i.e. `cfg` everywhere

File: benchmarks/proto/test.pb.go [hunk 3]
```diff
@@ -416,16 +416,16 @@ func (m *ResultStatus) MarshalTo(data []byte) (int, error) {
 	}
 	if m.PubKey == nil {
 		return 0, github_com_gogo_protobuf_proto.NewRequiredNotSetError("pubKey")
-	} else {
```
Comment (melekes) [inline benchmarks/proto/test.pb.go]:
```
// Code generated by protoc-gen-gogo.
// source: test.proto
// DO NOT EDIT!
```

File: types/priv_validator/sign_info.go (hunk 1 above)
Comment (tomc974) [reply types/priv_validator/sign_info.go]: Nothing's wrong with `lsi`, it's just that some receivers for `LastSignedInfo` were named `lsi` and others were named `info`. I just settle for `info` for all of them. I will switch to `lsi` instead.

Comment (melekes) [reply types/priv_validator/sign_info.go]: got it. thanks

File: config/config.go (hunk 2 above)
Comment (tomc974) [reply config/config.go]: `BaseConfig` type was using `b` and `c` as receiver names and `ConsensusConfig` type was using `c` and `cfg`. I tried to minimize changes by settling on `c` for `BaseConfig` and on `cfg` for `ConsensusConfig`. I will change to `cfg` for both types.

File: benchmarks/proto/test.pb.go (hunk 3 above)
Comment (tomc974) [reply benchmarks/proto/test.pb.go]: Oh, I miss that! I will revert the changes...

**Comment by tomc974** [comment]
@melekes All comments should be fixed now.

**Comment by melekes** [comment]
🥇 🏎 🍰

**Pull request merged**